src/Cafe/IOSU/legacy/iosu_fpd.cpp

```diff
@@ -165,9 +165,24 @@ namespace iosu
 				g_fpd.myPresence.isOnline = 1;
 				g_fpd.myPresence.gameKey.titleId = CafeSystem::GetForegroundTitleId();
 				g_fpd.myPresence.gameKey.ukn = CafeSystem::GetForegroundTitleVersion();
+
+				// Resolve potential domain to IP address
+				struct addrinfo hints = {0}, *addrs;
+				hints.ai_family = AF_INET;
+
+				const int status = getaddrinfo(nexTokenResult.nexToken.host, NULL, &hints, &addrs);
+				if (status != 0) {
+					cemuLog_log(LogType::Force, "IOSU_FPD: Failed to resolve hostname {}, {}", nexTokenResult.nexToken.host, gai_strerror(status));
+					return;
+				}
+
+				char addrstr[NI_MAXHOST];
+				getnameinfo(addrs->ai_addr, addrs->ai_addrlen, addrstr, sizeof addrstr, NULL, 0, NI_NUMERICHOST);
+				cemuLog_log(LogType::Force, "IOSU_FPD: Resolved IP for hostname {}, {}", nexTokenResult.nexToken.host, addrstr);
+
 				// start session
 				uint32 hostIp;
-				inet_pton(AF_INET, nexTokenResult.nexToken.host, &hostIp);
+				inet_pton(AF_INET, addrstr, &hostIp);
 				g_fpd.nexFriendSession = new NexFriends(hostIp, nexTokenResult.nexToken.port, "ridfebb9", myPid, nexTokenResult.nexToken.nexPassword, nexTokenResult.nexToken.token, accountId, (uint8*)&miiData, (wchar_t*)screenName, (uint8)countryCode, g_fpd.myPresence);
```

Comment:
Since you already have the address in a `struct addrinfo`, you could directly use that instead re-converting `addrstr` with `inet_pton()`.

Also, the `struct addrinfo` returned by `getaddrinfo()` should be freed with `freeaddrinfo()` once you don't need it anymore.

Hence, you could do something like this:

```cpp
const uint32_t hostIp = ((struct sockaddr_in*)addrs->ai_addr)->sin_addr.s_addr;
freeaddrinfo(addrs);
```

Thanks for your patience 😅

Reply:
No no, thank *you* for your patience